From a41e8821c5463e0c80e3826bf3cee74310360c9c Mon Sep 17 00:00:00 2001
From: warped-rudi <r.ihle@s-t.de>
Date: Thu, 6 Aug 2015 08:38:00 +0200
Subject: [PATCH 5/5] video: mxc_hdmi: Flag video modes from the fallback list
 as standard.

This patch changes the fallback modes from 'User' to 'Standard' (i.e.
they get listed as S:xxxx instead of U:xxxx).
---
 drivers/video/mxc/mxc_hdmi.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/video/mxc/mxc_hdmi.c b/drivers/video/mxc/mxc_hdmi.c
index ab9625f..353447b 100644
--- a/drivers/video/mxc/mxc_hdmi.c
+++ b/drivers/video/mxc/mxc_hdmi.c
@@ -1957,8 +1957,11 @@ static void  mxc_hdmi_default_modelist(struct mxc_hdmi *hdmi)
 	/*Add all no interlaced CEA mode to default modelist */
 	for (i = 0; i < ARRAY_SIZE(mxc_cea_mode); i++) {
 		mode = &mxc_cea_mode[i];
-		if (!(mode->vmode & FB_VMODE_INTERLACED) && (mode->xres != 0))
-			fb_add_videomode(mode, &hdmi->fbi->modelist);
+		if (!(mode->vmode & FB_VMODE_INTERLACED) && (mode->xres != 0)) {
+			struct fb_videomode m = *mode;
+			m.flag |= FB_MODE_IS_STANDARD;
+			fb_add_videomode(&m, &hdmi->fbi->modelist);
+		}
 	}
 
 	console_unlock();
-- 
1.7.9.5

